sky/unit/test/engine/mock_events.dart

```diff
@@ -79,3 +81,71 @@ class TestGestureEvent extends sky.GestureEvent {
   double velocityX;
   double velocityY;
 }
+
+class _PointerState {
+  bool isDown = false;
+  sky.Point location;
+}
+
+class TestPointerGenerator {
```

Comment:
Or just have the TestPointer constructor take the ID explicitly and leave it up to the caller. The part of my suggestion I care about is having self-contained objects representing the pointers to generate the events, rather than a generator object with hidden state objects representing the pointers.